src/main/java/org/junitpioneer/jupiter/StdIOExtension.java

```diff
@@ -27,39 +29,42 @@
 	private static final String SEPARATOR = System.getProperty("line.separator");
 
 	@Override
-	public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext)
-			throws ParameterResolutionException {
-		return parameterContext.isAnnotated(Std.class);
+	public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
+		return parameterContext.isAnnotated(StdIntercept.class);
 	}
 
 	@Override
-	public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext)
-			throws ParameterResolutionException {
+	public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
 		Class<?> parameterType = parameterContext.getParameter().getType();
+		StdIntercept stdIntercept = getInterceptAnnotation(parameterContext);
 		if (parameterType == StdOut.class) {
 			return new StdOut();
 		}
 		if (parameterType == StdIn.class) {
-			Std std = parameterContext
-					.findAnnotation(Std.class)
-					.orElseThrow(() -> new ParameterResolutionException("Needs to be annotated with Std"));
-			return new StdIn(std.value());
+			return new StdIn(stdIntercept.value());
 		}
 		throw new ParameterResolutionException("Can only resolve parameter of type " + StdOut.class.getName() + " or "
 				+ StdIn.class.getName() + " but was: " + parameterType.getName());
 
 	}
 
+	private StdIntercept getInterceptAnnotation(ParameterContext context) {
+		return context
+				.findAnnotation(StdIntercept.class)
+				.orElseThrow(() -> new ParameterResolutionException(
+					"This can never happen because this extension only supplies parameters if the parameter is annotated with @StdIntercept"));
+	}
+
```

Comment:
Shall we move this to the `PioneerAnnotationUtils` in a generic way? Is it even possible?

Reply:
You *could* but it does not really make sense, because this `ParameterResolutionException` is specific to this extension. If you want to move just the first part, how would that even look?
`PioneerAnnotationUtils.findAnnotation(context, StdIntercept.class)`?

I think this is simple enough to leave as-is but if you have any ideas, I'm open.

Reply:
> You _could_ but it does not really make sense, because this `ParameterResolutionException` is specific to this extension. If you want to move just the first part, how would that even look?
> `PioneerAnnotationUtils.findAnnotation(context, StdIntercept.class)`?
> 
> I think this is simple enough to leave as-is but if you have any ideas, I'm open.


My thouhgt was to include the exception throwing when moving. Does not have to be `ParameterResolutionException`, but maybe something "general" like `InvalidParameterException` or `MissingResourceException` - or we define a  `PioneerException` :D